packages/monitor-v2/src/monitor-polymarket/MonitorProposalsOrderBook.ts

```diff
@@ -202,27 +202,26 @@ export async function monitorTransactionsProposedOrderBook(
   const proposals = [...proposalsV2, ...proposalsV1];
 
   console.log(`Checking proposal price for ${proposals.length} markets...`);
-  const notifiedProposals = [];
-
-  for (const proposal of proposals) {
-    // Skip proposals that have already been processed.
-    if (Object.keys(pastNotifiedProposals).includes(getProposalKeyToStore(proposal))) {
-      continue;
-    }
-
-    try {
-      // Determine version based on which proposals array this came from
-      const version = proposalsV2.includes(proposal) ? "v2" : "v1";
-      const { notified } = await processProposal(proposal, params, logger, version);
-      if (notified) {
-        notifiedProposals.push(proposal);
+  await Promise.all(
+    proposals.map(async (proposal) => {
+      // Skip if we have already handled this proposal
+      if (Object.keys(pastNotifiedProposals).includes(getProposalKeyToStore(proposal))) {
+        return;
       }
-    } catch (error) {
-      await logFailedMarketProposalVerification(logger, params.chainId, proposal, error as Error);
-      notifiedProposals.push(proposal);
-    }
-  }
 
-  await storeNotifiedProposals(notifiedProposals);
```

Comment:
why do we need to break apart and invoke `storeNotifiedProposals` both in try and catch blocks below? `storeNotifiedProposals` is already doing paralel processing when writing to the datastore, so we could leave that at the end of processing all proposals, same as before?

Reply:
Function wise, both approaches persist exactly the same proposals, those that notified or errored, so external behaviour is identical.
Calling `storeNotifiedProposals` immediately just does it sooner (for crash safety), not differently.